site/cypress/e2e/Cypress-Feature/office_hours_queue.spec.js

```diff
@@ -83,86 +95,118 @@ const editAnnouncement = (text = '') => {
 };
 
 describe('test office hours queue', () => {
-    it('Creating queues and changing queue codes', () => {
+    before(() => {
         cy.login();
-        enableQueue();
-        // deleting the Lab help and homework debugging
-        deleteQueue();
-        deleteQueue();
+
+        // enable queue if not already enabled
+        cy.visit(['sample', 'config']); // course setting
+        cy.get('[data-testid="queue-enabled"]').check();
+        cy.get('[data-testid="queue-enabled"]').should('be.checked');
+        cy.reload();
+
         openNewQueue(queueName, queueCode);
-        openNewQueue(queueName, queueCode1); // same name but used different code
+        cy.get('[data-testid="popup-message"]').should('contain', 'New queue added');
+        openNewQueue(queueName_random, 'RANDOM');
+        cy.get('[data-testid="popup-message"]').should('contain', 'New queue added');
+        openNewQueue(queueName_blank, '');
+        cy.get('[data-testid="popup-message"]').should('contain', 'New queue added');
+    });
+
+    it('Creating queues and changing queue codes', () => {
+        cy.login();
+        // Create queue with same name but different code to ensure error
+        openNewQueue(queueName, queueCode1);
         cy.get('[data-testid="popup-message"]').should('contain', 'Unable to add queue. Make sure you have a unique queue name');
 
-        openNewQueue(queueName_random, 'RANDOM');
         changeQueueCode(queueName_random, 'RANDOM');
         cy.get('[data-testid="popup-message"]').should('contain', 'Queue Access Code Changed');
 
-        openNewQueue(queueName_blank, '');
         changeQueueCode(queueName_blank, '');
         cy.get('[data-testid="popup-message"]').should('contain', 'Queue Access Code Changed');
 
+        // Change 1st queue's code
         changeQueueCode(queueName, newQueueCode);
         cy.get('[data-testid="popup-message"]').should('contain', 'Queue Access Code Changed');
-
-        openNewQueue(queueName1, queueCode1);
     });
+
     it('Joining queues as student', () => {
         // switch to student to join queue
-        switchUser('student');
+        switchUser('bitdiddle');
         cy.visit(['sample', 'office_hours_queue']);
```

Comment:
I don't think this is needed now

Reply:
This test refactored a lot of it to use bitdiddle as the student

Reply:
Alex (author) said the reason was that student was already on the queue, while bitdiddle was not

Reply:
I understand the switch from student to bitdiddle, its the `cy.visit(['sample', 'office_hours_queue']);` part that is not needed, as the switchUser function already does `cy.visit(['sample', 'office_hours_queue']);`